src/js/components/Button.jsx

```diff
@@ -1,42 +1,114 @@
+    if ($disabled) {
+      return THEME.buttonDisabled.backgroundColor;
+    } else if ($secondaryButton) {
+      return $active
+        ? THEME.secondaryButtonSelected.backgroundColor
+        : THEME.secondaryButtonDefault.backgroundColor;
+    } else {
+      return $active
+        ? THEME.primaryButtonSelected.backgroundColor
+        : THEME.primaryButtonDefault.backgroundColor;
+    }
+  }};
+  border-color: ${({ $active, $disabled, $secondaryButton }) => {
+    if ($disabled) {
+      return THEME.buttonDisabled.borderColor;
+    } else if ($secondaryButton) {
+      return $active
+        ? THEME.secondaryButtonSelected.borderColor
+        : THEME.secondaryButtonDefault.borderColor;
+    } else {
+      return $active
+        ? THEME.primaryButtonSelected.borderColor
+        : THEME.primaryButtonDefault.borderColor;
+    }
+  }};
+  border: 1px solid;
   border-radius: 0.25rem;
-  color: ${getColor};
-  font-weight: 700;
-  font-size: 1rem;
+  color: ${({ $active, $disabled, $secondaryButton }) => {
+    if ($disabled) {
+      return THEME.buttonDisabled.fillColor;
+    } else if ($secondaryButton) {
+      return $active
+        ? THEME.secondaryButtonSelected.fillColor
+        : THEME.secondaryButtonDefault.fillColor;
+    } else {
+      return $active ? THEME.primaryButtonSelected.fillColor : THEME.primaryButtonDefault.fillColor;
+    }
+  }};
+  cursor: ${({ $disabled }) => ($disabled ? "not-allowed" : "pointer")};
+  font-style: var(--unnamed-font-family-roboto);
+  font-size: 18px;
+  font-weight: var(--unnamed-font-weight-medium);
+  letter-spacing: 0px;
   line-height: 1.5;
+  max-height: 37px;
+  padding: 0.2rem 0.5rem;
+  pointer-events: ${({ $disabled }) => ($disabled ? "none" : "all")}
   text-align: center;
   transition: color 0.15s ease-in-out, background-color 0.15s ease-in-out,
     border-color 0.15s ease-in-out;
-  padding: 0.2rem 0.5rem;
   vertical-align: middle;
 
   &:hover {
-    filter: brightness(90%);
+    background-color: ${({ $disabled, $secondaryButton }) => {
+      if ($disabled) {
+        return THEME.buttonDisabled.backgroundColor;
+      } else if ($secondaryButton) {
+        return THEME.secondaryButtonHover.backgroundColor;
+      } else {
+        return THEME.primaryButtonHover.backgroundColor;
+      }
+    }};
+    border-color: ${({ $disabled, $secondaryButton }) => {
+      if ($disabled) {
+        return THEME.buttonDisabled.borderColor;
+      } else if ($secondaryButton) {
+        return THEME.secondaryButtonHover.borderColor;
+      } else {
+        return THEME.primaryButtonHover.borderColor;
+      }
+    }};
+    color: ${({ $disabled, $secondaryButton }) => {
+      if ($disabled) {
+        return THEME.buttonDisabled.fillColor;
+      } else if ($secondaryButton) {
+        return THEME.secondaryButtonHover.fillColor;
+      } else {
+        return THEME.primaryButtonHover.fillColor;
+      }
+    }};
   }
 `;
 
+function Button({
+  active,
+  buttonText,
```

Comment:
We don't need `buttonText` prop, we can use default `children` and then use it like `<Button>buttonText<Button>`